backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java

```diff
@@ -1988,19 +2065,50 @@ public void deleteExperimentRuns(
             "ExperimentRun IDs not found in DeleteExperimentRuns request");
       }
 
+      Map<String, Long> workspaceIdByExperimentRunId =
+          request.getIdsList().stream()
+              .collect(
+                  Collectors.toMap(
+                      id -> id,
+                      id ->
+                          roleService
+                              .getEntityResource(
+                                  experimentRunDAO.getProjectIdByExperimentRunId(id),
+                                  ModelDBServiceResourceTypes.PROJECT)
+                              .getWorkspaceId()));
       List<String> deleteExperimentRunsIds =
           experimentRunDAO.deleteExperimentRuns(request.getIdsList());
       DeleteExperimentRuns.Response response =
           DeleteExperimentRuns.Response.newBuilder()
               .setStatus(!deleteExperimentRunsIds.isEmpty())
               .build();
-      saveAuditLogs(
-          Optional.empty(),
-          ModelDBServiceActions.DELETE,
-          deleteExperimentRunsIds,
-          ModelDBUtils.getStringFromProtoObject(request),
-          ModelDBUtils.getStringFromProtoObject(response),
-          null);
+      Function<MessageOrBuilder, String> getStringFromProtoObject =
+          messageOrBuilder -> {
+            try {
+              return ModelDBUtils.getStringFromProtoObject(messageOrBuilder);
+            } catch (InvalidProtocolBufferException e) {
+              throw new ModelDBException(e);
+            }
+          };
+      List<AuditLogLocalEntity> auditLogLocalEntities =
+          workspaceIdByExperimentRunId.entrySet().stream()
+              .map(
+                  entry ->
+                      new AuditLogLocalEntity(
+                          SERVICE_NAME,
+                          authService.getVertaIdFromUserInfo(authService.getCurrentLoginUserInfo()),
+                          ModelDBServiceActions.DELETE,
+                          entry.getKey(),
+                          ModelDBServiceResourceTypes.EXPERIMENT_RUN,
+                          Service.MODELDB_SERVICE,
+                          MonitoringInterceptor.METHOD_NAME.get(),
+                          getStringFromProtoObject.apply(request),
+                          getStringFromProtoObject.apply(response),
+                          entry.getValue()))
```

Comment:
This seems like an anti-pattern as we're de-capsulating the method `saveAuditLogs`. Is there a reason why we can't just use it?

Reply:
cc @evgeniilezhnin

Reply:
FIxed